### PR TITLE
Adds support for Azure AD

### PR DESCRIPTION
**Pull request by primetheus**
This PR adds support for Azure AD.

### Authentication methods
- [ ] Username/Password
- [x] Service Principal
- [ ] Certificate
- [ ] Device Auth

---
To use AzureAD, simply change `USER_DIRECTORY` to `AAD` and set the Azure settings

```env
## User directory to sync GitHub teams from
## Azure AD = AAD
## Active Directory = LDAP
## OpenLDAP = LDAP
USER_DIRECTORY=AAD

#######################
## Azure AD Settings ##
#######################
## Azure Tenant ID
AZURE_TENANT_ID="<tenant_id>"
## Azure Service Principal Client ID
AZURE_CLIENT_ID="<client_id>"
## Azure Service Principal Client Secret
AZURE_CLIENT_SECRET="<client_secret>"
## Azure AD directory scopes. If you have multiple directories, separate them by spaces
## i.e. "Directory.Read.all User.Read.all"
AZURE_APP_SCOPE=".default"
## API endpoint for Azure AD (Graph) group member queries
AZURE_API_ENDPOINT="https://graph.microsoft.com/v1.0"
## Custom attribute for usernames
USERNAME_ATTRIBUTE=userPrincipalName
```

---
This app requires the following Azure permissions:

- `Directory.Read.All`
- `Group.Read.All`
- `GroupMember.Read.All`
- `Organization.Read.All`
- `User.Read.All`

### CODE REVIEW

File: githubapp/azuread.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        member_list = []
+        # Calling graph using the access token
+        graph_data = requests.get(  # Use token to call downstream service
+            self.AZURE_API_ENDPOINT,
+            headers={'Authorization': f'Bearer {token}'}
+        ).json()
+        # print("Graph API call result: %s" % json.dumps(graph_data, indent=2))
+        groups = json.loads(json.dumps(graph_data, indent=2))
+        for group in groups['value']:
```
Comment (primetheus) [inline githubapp/azuread.py:96]: @stoe this currently pulls _all_ groups and iterates over them. We'll need to add a filter to lookup a group and return the ID.

File: githubapp/azuread.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            ).json()
+            for member in members['value']:
+                member_list.append(member[self.USERNAME_ATTRIBUTE])
+        return member_list
+
+
+if __name__ == '__main__':
+    aad = AzureAD()
+    token = aad.get_access_token()
+    aad.get_group_members(token=token, group="0aafe564-0044-424c-8a75-e8a59e6a83d5")
```
Comment (primetheus) [inline githubapp/azuread.py:109]: this section can be deleted when the group filter is added... it's just for testing

**Review by stoe — DISMISSED** [review]
LGTM

**Review by stoe — APPROVED** [review]

**Review by stoe — APPROVED** [review]
I'll give it a spin today.

**Review by stoe — CHANGES_REQUESTED** [review]
Should probably address the linter issues!?

**Review by stoe — APPROVED** [review]

**Pull request merged**